sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -311,6 +312,97 @@ def predict(self, X):
         return X[:, 0]
 
 
+class OneClassErrorClassifier(BaseBadClassifier):
+    def fit(self, X, y):
+        # Convert data
+        X, y = check_X_y(
+            X, y, accept_sparse=("csr", "csc"), multi_output=True, y_numeric=True
+        )
+        cls, y = np.unique(y, return_inverse=True)
+        # find the number of class after trimming
+        if cls.shape[0] < 2:
+            raise ValueError("Nonsensical Error")
+
+    def predict(self, X):
+        X = check_array(X)
+        return np.ones(X.shape[0])
+
+
+class OneClassSampleErrorClassifier(BaseBadClassifier):
```

Comment:
I do not think this estimator is being instantiated and tested in this file.

Reply:
Indeed, none of three new classes `OneClassErrorClassifier`, `OneClassSampleErrorClassifier` and `OneClassSampleErrorClassifierPredict` were instantiated and tested.
I have introduced a test for `OneClassSampleErrorClassifier`.
I have removed `OneClassErrorClassifier` because it seems to  me that there is nothing related to the weight issue than can be tested with it.
I have kept `OneClassSampleErrorClassifierPredict`, because I'm unsure it should be used to test predictions, but I would rather get rid of it... I guess I should just call `predict` on OneClassSampleErrorClassifier`, in order to test errors on predictions... am I wrong?

Reply:
I realized how `OneClassSampleErrorClassifierPredict` works... so I kept it and tested it.